app/src/main/java/com/mathewsachin/fategrandautomata/imaging/DroidCvPattern.kt

```diff
@@ -75,9 +92,23 @@ class DroidCvPattern(private var Mat: Mat? = Mat(), private val OwnsMat: Boolean
             val result = DisposableMat()
 
             if (Template.width <= width && Template.height <= height) {
-                Imgproc.matchTemplate(Mat, Template.Mat, result.Mat, Imgproc.TM_CCOEFF_NORMED)
-            }
-            else logd("Skipped matching $Template: Region out of bounds")
+                alpha.let { mask ->
+                    if (mask != null) {
+                        Imgproc.matchTemplate(
+                            Mat,
+                            Template.Mat,
+                            result.Mat,
+                            Imgproc.TM_CCOEFF_NORMED,
+                            mask
+                        )
+                    } else Imgproc.matchTemplate(
```

Comment:
To keep the if and else similar, I would add curly braces here too.

Reply:
done